src/lib/containers/oauth/CreateOAuthClient.tsx

```diff
@@ -85,16 +86,24 @@ export const CreateOAuthModal: React.FunctionComponent<
   const hideConfirmModal = () => {
     setIsShowingConfirmModal(false)
     setIsDelete(false)
-    setSelectedClient(undefined)
   }
 
-  const { mutate } = useMutateOAuthClient({
+  const { mutate: createClient } = useCreateOAuthClient({
     onSuccess: () => {
-      displayToast(`Successfully saved`, 'success')
-      hide()
+      displayToast('Successfully created', 'success')
     },
-    onError: (err: any) => {
-      displayToast(err.reason as string, 'danger')
+    onError: error => {
+      displayToast(error.reason as string, 'danger')
+    },
```

Comment:
For create/edit, instead of pushing a toast, can you store the error in state and show it in the modal in a bootstrap `Alert`? The response in these cases is usually a form error, and we should keep that on the screen with the form instead of showing it in an ephemeral toast.